### ISSUE TITLE
Consider setting `production` mode for webpack config

### ISSUE THREAD

**Issue by romansp** [skpm/skpm#159]
https://github.com/skpm/skpm/blob/0050accbcc7a9f76d8d1454e3e0714c52f46194c/packages/builder/src/utils/webpackConfig.js#L126-L127

I don't know whether it was intentional or not , but setting [webpack's mode](https://webpack.js.org/concepts/mode/) to 'production' (when `NODE_ENV == 'production'`) will result in more efficient builds by default.

**Comment by mathieudutour** [comment]
IIRC, when I looked at it, setting the mode to `production` add an `Uglify` plugin. The issue is that we need a special option for uglify:
```js
      new Uglify({
          uglifyOptions: {
            mangle: {
              // @see https://bugs.webkit.org/show_bug.cgi?id=171041
              // @see https://github.com/mishoo/UglifyJS2/issues/1753#issuecomment-324814782
              safari10: true,
            },
          },
        })
```

And I don't think I found a way to override it. But maybe I didn't look in the right place

**Comment by romansp** [comment]
Okay, at least I can see setting `devtool` to `'none'` or `'source-map'` as well for `isProd` case. 

Right now `undefined` fallbacks webpack to `eval` devtool mode. Emitted JS is at least 2 times bigger than with `none` for default `skpm` plugin template.

**Comment by mathieudutour** [comment]
oh waow, I would have expected `undefined` to be `'none'`... Changing that right away

**Comment by mathieudutour** [comment]
https://github.com/skpm/skpm/commit/92cdb68546a23aa10c93238837ef7aaa3b636bb0

**Comment by romansp** [comment]
And for UglifyJsOptions. 

It's possible to override options in `'production'` mode without adding additional plugin. Here are the [docs](https://webpack.js.org/configuration/optimization/#optimization-minimizer) and [sample](https://github.com/damianobarbati/yarsk/blob/50b6f352a13ec2e778fa8b252f915550b6132964/config/webpack.config.js#L110).

**Comment by mathieudutour** [comment]
fixed in 0.6
